app/assets/javascripts/location_search.js

```diff
@@ -2,26 +2,60 @@ window.ST = window.ST || {};
 
 (function(module) {
 
-  module.initializeLocationSearch = function() {
+  module.initializeLocationSearch = function(minimumDistanceMax) {
     var searchInput = document.getElementById('q');
     var statusInput = document.getElementById('ls');
     var coordinateInput = document.getElementById('lc');
     var boundingboxInput = document.getElementById('boundingbox');
+    var maxDistanceInput = document.getElementById('distance_max');
     var homepageForm = document.getElementById('homepage-filters');
     var autocomplete = new window.google.maps.places.Autocomplete(searchInput, { bounds: { north: -90, east: -180, south: 90, west: 180 } });
     autocomplete.setTypes(['geocode']);
 
     boundingboxInput.value = null;
 
+    function toRadians(degrees) {
+      return degrees * (3.1415/180);
+    }
+
+    function computeScale(a, b) {
+      var R = 6371; // Earth's radius in km
+
+      var lat1 = a.lat();
+      var lat2 = b.lat();
+      var lng1 = a.lng();
+      var lng2 = b.lng();
+      var φ1 = toRadians(lat1);
+      var φ2 = toRadians(lat2);
+      var Δφ = toRadians(lat2-lat1);
+      var Δλ = toRadians(lng2-lng1);
+
+      var a = Math.sin(Δφ/2) * Math.sin(Δφ/2) +
+              Math.cos(φ1) * Math.cos(φ2) *
+              Math.sin(Δλ/2) * Math.sin(Δλ/2);
+      var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a));
```

Comment:
If the above (and below) code is some known algorithm, a link would help digesting this.